app/bin/databricks_common_utils.py

```diff
@@ -255,66 +291,229 @@ def format_to_json_parameters(params):
+                        "Check logs for more details.".format(str(resp.status_code))
+                    )
             else:
-                msg = 'Unable to validate Azure Active Directory Credentials. Check logs for more details.'
+                msg = (
+                    "Unable to request Databricks instance. "
+                    "Please validate the provided Databricks and "
+                    "Proxy configurations or check the network connectivity."
+                )
                 _LOGGER.error("Error while trying to generate AAD access token: {}".format(str(e)))
                 _LOGGER.debug(traceback.format_exc())
             _LOGGER.error(msg)
             if retry == 0:
                 return msg, False
+
+
+def check_user_roles(session_key, validate=None):
+    """Method to check user roles."""
+    required_role = const.REQUIRED_ROLES
+    searchquery_oneshot = "| rest /services/authentication/current-context splunk_server=local | table roles"
+    kwargs_oneshot = {
+        "output_mode": 'json'
+    }
+    roles = []
+    try:
+        mgmt_port = cli.getMgmtUri().split(":")[-1]
+        service = client.connect(
+            port=mgmt_port,
+            token=session_key,
+            app=APP_NAME
+        )
+    except Exception as e:
+        _LOGGER.error(
+            "Databricks Error: Error while connecting to"
+            " splunklib client. Error: " + str(e))
+        _LOGGER.debug(
+            "Databricks Error: Error while connecting to"
+            " splunklib client. Error: " + traceback.format_exc())
+    try:
+        oneshotsearch_results = service.jobs.oneshot(searchquery_oneshot, **kwargs_oneshot)
+
+        # Get the results and display them using the JSONResultsReader
+        reader = results.JSONResultsReader(oneshotsearch_results)
+        for item in reader:
+            if isinstance(item, dict) and item.get("roles"):
+                values = item.get("roles", None)
+                if type(values) == list:
+                    roles = values
+                else:
+                    roles.append(values)
+        if not roles:
+            raise Exception("No Role found.")
+        if not validate:
+            if set(roles) & set(required_role):
+                return True
+            else:
+                return False
+        else:
+            if "databricks_admin" not in set(roles):
+                return False
+            else:
+                return True
+    except Exception as e:
+        _LOGGER.error(
+            "Databricks Error: Error while fetching"
+            " logged in user roles. Error: " + str(e))
+        _LOGGER.debug(
+            "Databricks Error: Error while fetching"
+            " logged in user roles. Error: " + traceback.format_exc())
+
+
+def get_user_agent(session_key):
+    """Method to get user agent."""
+    current_user = get_current_user(session_key)
+    return "{}-{}".format(const.USER_AGENT_CONST, current_user)
```

Comment:
you are not allowed to add the current user to the user agent, because it's PII. User-agent cannot contain PII.

Reply:
It was the primary requirement stated by Zafer for this release to include Splunk's logged-in user name in the user header. We can discuss this with Zafer and revert the changes.